caikit/runtime/servicers/training_management_servicer.py

```diff
@@ -45,9 +45,14 @@ def GetTrainingStatus(self, request, context):  # pylint: disable=unused-argumen
                 training_id=training_info.training_id
             )
 
+            reasons = []
+            if model_future.get_info().errors:
+                reasons = [str(error) for error in model_future.get_info().errors]
```

Comment:
nit that `CodeQL` is gonna barf on this since `str(exception)` could possibly include stack trace info. See this: https://github.com/caikit/caikit/security/code-scanning/4

I think we may need something a little bit more fancy here to grab just the relevant error message off of the exception. We do a pretty good job in the inference stack of catching error cases and translating to appropriate `CaikitRuntimeException`s, maybe the training stack needs a bit more love as well?

I could be fine with merging this since I think stack trace exposure of open source software isn't extremely bad, but we should think on this